### DISCUSSION TITLE
Sub sections like gaming

### DISCUSSION THREAD

**Discussion by Raphty** [portmaster-community/apps#3] [New Profiles]
Categories like Gaming can be useful, what sub sections in this Repo would you also find useful?

I for instance don't particularly like having too many sections, but I understand that some help finding things easier.

**Comment by innovatodev** [comment]
My dream would be something like that to be honest:

```plaintext
/Generic
    /GlobalSettings
        /PromptForNewProfiles (Prompt when a new app needs internet!)
            README.MD (Informations, HowTo, etc.)
            Profile.yml
        /ForceBlockThenPromptOrAllow (Force block when a new app needs internet, disabling the force block when need internet!)
            README.MD (Informations, HowTo, etc.)
            Profile.yml
    /Templates|UseCases
        /WebBrowser
            README.MD (Informations, HowTo, etc.)
            Profile.yml
        /Email
            README.MD (Informations, HowTo, etc.)
            Profile.yml
    /AppsProfiles (Anything here does not contains any fingerprint to be crossplatform)
        /Firefox (Firefox!)
            README.MD (Informations, HowTo, etc.)
            Profile.yml
        /FirefoxHardened (Firefox hardened to allow only HTTPS port!)
            README.MD (Informations, HowTo, etc.)
            Profile.yml
/Windows
    /GlobalSettings
        /MaybeSomePlatformSpecificGlobals
            README.MD (Informations, HowTo, etc.)
            Profile.yml
    /AppsProfiles
        /MullvadWireguardOwned (Wireguard, Mullvad owned servers ONLY!)
            README.MD (Please set your custom DNS to 127.0.0.1 in the client!)
            Profile.yml
        /MullvadWireguardRented (Wireguard, all rented servers included!)
            README.MD (Please set your custom DNS to 127.0.0.1 in the client!)
            Profile.yml
        /MullvadOpenVPNOwned (OpenVPN, Mullvad owned servers ONLY!)
            README.MD (Please set your custom DNS to 127.0.0.1 in the client!)
            AllInOne.yml (Include both fingerprint in one profile)
            MullvadDaemon.yml (The Daemon)
            OpenVPN.yml (OpenVPN.exe)
        /SteamAndGames (Steam + all games in a single profile!)
            README.MD (Informations, HowTo, etc.)
            Profile.yml
/Linux (Could even split into more subdirectories like Ubuntu based, Redhat based etc.)
    /GlobalSettings
        /MaybeSomePlatformSpecificGlobals
            README.MD (Informations, HowTo, etc.)
            Profile.yml
    /AppsProfiles
        /KDEConnect
            README.MD (Informations, HowTo, etc.)
            Profile.yml
```

This structure would allow community to:

- Create generic GlobalSettings + AppsProfiles + Templates
- Create System Specific GlobalSettings + AppsProfiles
- Anything would have a README.md with informations, tutorials dedicated to it.
- A community profile could potentially include more than 1 portmaster profile, simple exemples
    1. Allow updates on windows, profile containing all fingerprints in one profile and also differents profiles for Dosvc, wuauctl, BITS, etc.
    2. Block telemetry and reporting, profile containing all fingerprints in one profile and also differents profiles for WdiSystemHost, WdiServiceHost, DPS, DiagTrack, etc.
    3. Mullvad on wireguard uses a single daemon, but if openvpn, it then uses OpenVPN.exe (there hundred of different examples)
- Would be more easy to link the exact profile an user needs if there more splitting, some software uses different ways to do his jobs if on linux or windows, or even differently if ubuntu or redhat.
- If a know software only works if also managed with AnotherConnection or NetworkNoise profile, community could add it inside of the dedicated profile, like App1.yml, NetworkNoise.yml, AnotherConnection.yml to merge them easily with their configuration.
- Since users do not use their apps the same way, it would be easy to include multiples profiles for the same app, while keeping a clear explanation with a README for each of them, not just only a .yml thrown inside of a single AppsProfiles with just the filename to describe
- Can provide both system agnostic profiles for apps but also system specific fingerprinted profiles  ^^

**Reply by Raphty** [reply]
**Create generic GlobalSettings + AppsProfiles + Templates**

yes this is the idea for the use cases: https://github.com/Portmaster-Community/apps/tree/main/use-cases

we opted to call it use cases, so people go looking in there for a use case... like email, browser...

But I get generic as the name as well.
I am looking for a "Generic" profile...

I would love some more feedback on this here, so we can do this before more profiles come in :D

**Reply by Raphty** [reply]
**Create System Specific GlobalSettings + AppsProfiles**

I agree on the global settings here, but not on the specific app profiles, I don't think that the settings are super different depending on the platform they are on... so having one authoritative profile would be easier to find.

**Reply by Raphty** [reply]
**Anything would have a README.md with informations, tutorials dedicated to it.**
This I like, it would make the trees deeper but you are right it gives the option to explain what and why... I do like if there are comments in the settings as well... as seen here https://github.com/Portmaster-Community/apps/blob/main/use-cases/email-generic.yml this does work and your Portmaster will add this comment to the setting. so people see it in Portmaster as well.

**Reply by Raphty** [reply]
**A community profile could potentially include more than 1 portmaster profile, simple exemples**

cool idea, I don't yet see how this would work... but we will figure this out together😁

**Reply by Raphty** [reply]
**Would be more easy to link the exact profile an user needs if there more splitting, some software uses different ways to do his jobs if on linux or windows, or even differently if ubuntu or redhat.**

you mean because of the redme? you can link other profiles?
regarding windows and linux... again, I belive the settings will be super similar and having one easy profile will result in faster updates if the app needs new connections or some can get removed

**Reply by Raphty** [reply]
**If a know software only works if also managed with AnotherConnection or NetworkNoise profile, community could add it inside of the dedicated profile, like App1.yml, NetworkNoise.yml, AnotherConnection.yml to merge them easily with their configuration.**

At the moment settings get replaced... so if you have rules in the network noise and you import a new one, all old rules will be replaced by the imported ones... so this would have to be mentioned in the readme.

**Reply by Raphty** [reply]
**Since users do not use their apps the same way, it would be easy to include multiples profiles for the same app, while keeping a clear explanation with a README for each of them, not just only a .yml thrown inside of a single AppsProfiles with just the filename to describe**

I agree, but also adding comments to settings can help a lot.

**Reply by Raphty** [reply]
**Can provide both system agnostic profiles for apps but also system specific fingerprinted profiles ^^**

this was what my initial comment was about, we would like to collect all fingerprints in one so you can apply it on all machines.

**Reply by innovatodev** [reply]
> A community profile could potentially include more than 1 portmaster profile

Some softwares needs to play with another connection or network noise profile (a lot of things involving incoming connections needs some tweaks inside of network noise from what i see) so even if they can't merge without overwrite, they can at least see what they need to do that manually for those specials

Some others needs more than 1 executable to work, so we could provide 1 profile include all the necessary fingerprints in one profile, but also a set of different profiles for it (like portmaster do by default) so people have choice doing the normal way, or merging different "base profiles with 1 fingerprint into one bigger" :D

> this was what my initial comment was about, we would like to collect all fingerprints in one so you can apply it on all machines.

Oh yeah i didnt think about that but we can just make one fingerprint for windows and one for linux or even differents ditros, in the same profile afterall ^^ it changes a lot of think !!!

I also didnt noticed there commentary inside of the profile itself, so we can include just 1 app per root directory, and throw all sort of profile for it, with a readme covering concise description of each of them, and most of the technical informations directly inside of the .yml then ^^

So basically the only change would be a directory structure per application, all the rest would stay the same ? 🥰

**Reply by innovatodev** [reply]
So it could be something like that, everything from an app inside of a folder, and there can be differents usecases about each of those apps

```plaintext
/use-cases (No need to readme on those since it's made to be simple generic templates and its commented inside)
    WebBrowser.yml
    MailClient.yml
/global-defaults
    /PromptForNewProfiles
        README.MD (Prompt when a new app needs internet!)
        Profile.yml
    /ForceBlockThenPromptOrAllow
        README.MD (Force block when a new app needs internet, disabling the force block when need internet!)
        Profile.yml
/app-profiles
    /Firefox
        README.MD (Firefox with just blocklists! it contains all fingerprints required for linuxes, and also all sort of installations (user or system wide) for windows!)
        Profile.yml
    /Mullvad
        MullvadAllInOne.yml (Include all required fingerprints in one profile) = It's easier to manage services when all dependencies are inside of a single portmaster profile !
        MullvadDaemon.yml (Mullvad Daemon)
        MullvadOpenVPN.yml (OpenVPN.exe fingerprint) = Like auto-created portmaster's profile
        OwnedAllInOne.yml (Include all required fingerprints in one profile) = It's easier to manage services when all dependencies are inside of a single portmaster profile !
        OwnedDaemon.yml (Mullvad owned servers ONLY!)
        OwnedOpenVPN.yml (OpenVPN.exe fingerprint) = Like auto-created portmaster's profile
        README.MD (Please set your custom DNS to 127.0.0.1 in the client!)
    /Steam
        README.MD (Informations, HowTo, etc.)
        SteamAndGames.yml (Steam + all games in a single profile! Still protected by dns blocklists for unity telemetry! No need to bother having a portmaster profile for EACH GAMES !)
        Steam.yml (Steam and all its services in one profile! do not include games!)
    /KDEConnect
        README.MD (Informations, HowTo, etc.)
        Profile.yml
```

**Comment by Raphty** [comment]
@innovatodev thank you for the feedback I did the changes.

the "advanced" idea now is that categories can be established easily as well, by going deeper into the folder structure. so if we get to 100+ KDE services we can move them into a KDE folder beneath the service folder an so on...

I am sorry for the PRs I broke with those changes, I don't want to do this too often but since it is so early I wanted to get this changed sooner rather then later

**Reply by innovatodev** [reply]
Seems good , well better now than later thats for sure ^^
